test/util/util.go

```diff
@@ -0,0 +1,346 @@
+	gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+	return clientset
+}
+
+func CreateOcmWorkClient(config *rest.Config) *ocmWorkClient.Clientset {
+	ocmWorkClient, err := ocmWorkClient.NewForConfig(config)
+	gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+	return ocmWorkClient
+}
+
+func CreateDynamicClient(config *rest.Config) *dynamic.DynamicClient {
+	dynamicClient, err := dynamic.NewForConfig(config)
+	gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+	return dynamicClient
+}
+
+func CreateNS(ctx context.Context, client *kubernetes.Clientset, name string) {
+	labels := make(map[string]string)
+	labels["app.kubernetes.io/name"] = "nginx"
+	nsObj := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: "",
+			Labels:    labels,
+		},
+		Status: corev1.NamespaceStatus{},
+	}
+	client.CoreV1().Namespaces().Create(ctx, nsObj, metav1.CreateOptions{})
+	gomega.Eventually(func() error {
+		_, err := client.CoreV1().Namespaces().Get(ctx, name, metav1.GetOptions{})
+		return err
+	}, timeout).Should(gomega.Succeed())
+}
+
+func Cleanup() {
+	var e, o bytes.Buffer
+	cmd := exec.Command("../common/cleanup.sh")
+	cmd.Stderr = &e
+	cmd.Stdout = &o
+	err := cmd.Run()
+	if err != nil {
+		fmt.Fprintf(ginkgo.GinkgoWriter, "%s", o.String())
+		fmt.Fprintf(ginkgo.GinkgoWriter, "%s", e.String())
+	}
+	gomega.Expect(err).To(gomega.Succeed())
+}
+
+func SetupKubestellar(releasedFlag string) {
+	var cmd *exec.Cmd
+	var e, o bytes.Buffer
+	if releasedFlag == "on" {
+		cmd = exec.Command("../common/setup-kubestellar.sh", "--released")
+	} else {
+		cmd = exec.Command("../common/setup-kubestellar.sh")
+	}
+	cmd.Stderr = &e
+	cmd.Stdout = &o
+	err := cmd.Run()
+	if err != nil {
+		fmt.Fprintf(ginkgo.GinkgoWriter, "%s", o.String())
+		fmt.Fprintf(ginkgo.GinkgoWriter, "%s", e.String())
+	}
+	gomega.Expect(err).To(gomega.Succeed())
+}
+
+func int32Ptr(i int32) *int32 {
+	return &i
+}
+
+func DeleteBindingPolicy(ctx context.Context, wds *ksClient.Clientset, name string) {
+	wds.ControlV1alpha1().BindingPolicies().Delete(ctx, name, metav1.DeleteOptions{})
+	gomega.Eventually(func() error {
+		_, err := wds.ControlV1alpha1().BindingPolicies().Get(ctx, name, metav1.GetOptions{})
+		return err
+	}, timeout).Should(gomega.Not(gomega.Succeed()))
+}
+
+func CreateBindingPolicy(ctx context.Context, wds *ksClient.Clientset, name string,
+	clusterSelector map[string]string, objectSelector map[string]string) {
+	bindingPolicy := ksapi.BindingPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: ksapi.BindingPolicySpec{
+			ClusterSelectors: []metav1.LabelSelector{
+				{MatchLabels: clusterSelector},
+			},
+			Downsync: []ksapi.DownsyncObjectTest{
+				{ObjectSelectors: []metav1.LabelSelector{
+					{MatchLabels: objectSelector},
+				}},
+			},
+		},
+	}
+	wds.ControlV1alpha1().BindingPolicies().Create(ctx, &bindingPolicy, metav1.CreateOptions{})
```

Comment:
I think error checking here would be helpful

Reply:
done